### PR TITLE
[moderator-words] support regular expressions for filters

### PR DESCRIPTION
**Pull request by dims**
- use regular expressions so we can do complex matching going forward
- start by case insensitive look up for the first filter (so we would catch mixed case - say "Guys")

Signed-off-by: Davanum Srinivas <davanum@gmail.com>

### CODE REVIEW

**Comment by dims** [comment]
tested the bits in go playground https://go.dev/play/p/qeBBCc6QOKI

File: slack-moderator-words/events.go [hunk 1]
```diff
@@ -115,7 +114,12 @@ func (h *handler) ServeHTTP(rw http.ResponseWriter, r *http.Request) {
 	if h.filters != nil {
 		for _, filter := range h.filters {
 			for _, word := range filter.Triggers {
-				if strings.Contains(event.Event.Text, word) {
+				matched, err := regexp.MatchString(word, event.Event.Text)
```
Comment (BenTheElder) [inline slack-moderator-words/events.go:117]: TODO: amortize compiling the regexes on startup instead of once per regex per event.

**Review by BenTheElder — COMMENTED** [review]
SGTM

This does have the potential to cost a bit more than say, specifically handling case-insensitivity though, and if we ever wanted to account for attempts to evade the bot we might want to instead generalize banning 13375P34K forms etc.

... I kinda doubt we're going to do that, given the bot currently just privately nudges the commenter, but throwing out the thought anyhow before we commit to regex.

/lgtm
/hold

**Comment by dims** [comment]
closing this for now, we can revive if needed.

**Comment by jberkus** [comment]
Yeah, I have some concerns about cost, given that this runs for every post on Slack.

**Pull request closed without merging**